File: src/app/conjugation/forms/passive-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {PassiveAffirmative} from "./passive-affirmative";
import {NonPastShortNegative} from "./non-past-short-negative";

export class PassiveNegative implements Conjugation {

    getConjugation(word: Word): (Word | undefined) {

        const originalType = word.wordType;
        const passive = new PassiveAffirmative().getConjugation(word);
        if (passive === undefined) {
            return undefined;
        }

        return new NonPastShortNegative().getConjugation(passive.changeType(WordType.IchidanVerb))?.changeType(originalType)
    }

    getTitle(): string {
        return "Passive, negative"
    }

}

File: backend/app/conjugation/forms/passive_negative.py
from ..core import Conjugation, Word, WordType
from .non_past_short_negative import NonPastShortNegative
from .passive_affirmative import PassiveAffirmative


class PassiveNegative(Conjugation):

    title = 'Passive, negative'
    settings_title = 'Negative'

    def conjugate(self, word: Word) -> Word | None:
        original_type = word.word_type
        passive = PassiveAffirmative().conjugate(word)
        if passive is None:
            return None

        negative = NonPastShortNegative().conjugate(
            passive.change_type(WordType.ICHIDAN_VERB)
        )
        if negative is None:
            return None

        return negative.change_type(original_type)
